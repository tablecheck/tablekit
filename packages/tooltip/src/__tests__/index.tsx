import { mount } from '@tablecheck/tablekit-enzyme';

import { Tooltip } from '../index';

beforeEach(() => {
  global.console.error = jest.fn();
});

const testContent = <div>Test</div>;

describe('Tooltip', () => {
  test('should be possible to create a component', () => {
    const wrapper = mount(
      <Tooltip content={testContent}>
        {(setTargetRef) => <div ref={setTargetRef}>Target</div>}
      </Tooltip>
    );
    expect(wrapper).not.toBeUndefined();
  });

  test('should correctly apply styling', () => {
    expect(
      mount(
        <Tooltip content={testContent}>
          {(setTargetRef) => <div ref={setTargetRef}>Target</div>}
        </Tooltip>
      )
    ).toMatchSnapshot();
  });

  test('should allow passing of custom className', () => {
    expect(
      mount(
        <Tooltip content={testContent} className="test-class">
          {(setTargetRef) => <div ref={setTargetRef}>Target</div>}
        </Tooltip>
      )
    ).toMatchSnapshot();
  });
});
