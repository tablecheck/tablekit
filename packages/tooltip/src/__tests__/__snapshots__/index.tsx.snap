// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should allow passing of custom className 1`] = `
.emotion-0 {
  position: relative;
}

<Tooltip
  className="test-class"
  content={
    <div>
      Test
    </div>
  }
  delay={300}
  popperProps={
    Object {
      "placement": "bottom",
    }
  }
>
  <Container
    className="test-class"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="test-class emotion-0 emotion-1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <Manager>
        <Reference>
          <div>
            Target
          </div>
        </Reference>
      </Manager>
    </div>
  </Container>
</Tooltip>
`;

exports[`Tooltip should correctly apply styling 1`] = `
.emotion-0 {
  position: relative;
}

<Tooltip
  content={
    <div>
      Test
    </div>
  }
  delay={300}
  popperProps={
    Object {
      "placement": "bottom",
    }
  }
>
  <Container
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="emotion-0 emotion-1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <Manager>
        <Reference>
          <div>
            Target
          </div>
        </Reference>
      </Manager>
    </div>
  </Container>
</Tooltip>
`;
